packages/devtools-reps/src/reps/tests/promise.js

```diff
@@ -0,0 +1,208 @@
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test Promise fulfilled with array", () => {
+  const object = stubs.get("FulfilledWithArray");
+  const defaultOutput = `Promise { <state>: "fulfilled", <value>: [3] }`;
+
+  it("correctly selects PromiseRep Rep for Promise fulfilled with an array", () => {
+    expect(getRep(object)).toBe(PromiseRep.rep);
+  });
+
+  it("should render as expected", () => {
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "fulfilled" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test on DOM node MouseOver", () => {
+  const stub = stubs.get("FulfilledWithNode");
+  const grips = getSelectableInInspectorGrips(stub);
+
+  it("has one node grip", () => {
+    expect(grips.length).toBe(1);
+  });
+
+  it("calls with expected argument when mouseover is fired on the node element", () => {
+    let mouseOverValue;
+    const onDOMNodeMouseOver = (object) => {
+      mouseOverValue = object;
+    };
+
+    const renderedComponent = shallow(Rep({
+      object: stub,
+      onDOMNodeMouseOver,
+    }));
+
+    renderedComponent.find(".objectBox-node").simulate("mouseover");
+    expect(mouseOverValue).toEqual(grips[0]);
+  });
+});
+
+describe("test on DOM node MouseOut", () => {
+  const stub = stubs.get("FulfilledWithNode");
+  const grips = getSelectableInInspectorGrips(stub);
+
+  it("has one node grip", () => {
+    expect(grips.length).toBe(1);
+  });
+
+  it("calls with expected argument when mouseout is fired on the node", () => {
+    let called = false;
+    const onDOMNodeMouseOut = (object) => {
+      called = true;
+    };
+
+    const renderedComponent = shallow(Rep({
+      object: stub,
+      onDOMNodeMouseOut,
+    }));
+
+    renderedComponent.find(".objectBox-node").simulate("mouseout");
+    expect(called).toBe(true);
+  });
+});
+
+describe("test on DOM inspect icon click", () => {
+  const stub = stubs.get("FulfilledWithNode");
+  const grips = getSelectableInInspectorGrips(stub);
+  let inspectIconClickedValues = null;
+  const onInspectIconClick = (object) => {
+    inspectIconClickedValues = object;
+  };
+  const renderedComponent = shallow(Rep({
+    object: stub,
+    onInspectIconClick,
+  }));
+
+  it("has one node grip", () => {
+    expect(grips.length).toBe(1);
+  });
+
+  it("renders no inspect icon when the node is not connected to the DOM tree", () => {
+    const renderedComponentWithoutInspectIcon = shallow(Rep({
+      object: stubs.get("FulfilledWithDisconnectedNode"),
+      onInspectIconClick,
+    }));
+    const node = renderedComponentWithoutInspectIcon.find(".open-inspector");
+
+    expect(node.exists()).toBe(false);
+  });
+
+  it("renders an inspect icon", () => {
```

Comment:
this block is a bit weird here, I have trouble finding to which `renderedComponent` it applies.
Could you move it into the `it` block where we declare `renderedComponent` ? Or recreate a new component so we have it in the same block.
Thanks !